dev/com.ibm.ws.jsp.2.3_fat/fat/src/com/ibm/ws/jsp23/fat/tests/JSPChannelTest.java

```diff
@@ -0,0 +1,202 @@
+                             "Host: " + address + "\r\n" +
+                             "Keep-Alive: timeout=5, max=200\r\n" +
+                             "\r\n";
+
+            String redirect_request = "GET /" + APP_NAME + "/page2.jsp " + "HTTP/1.1\r\n" +
+                                      "Host: " + address + "\r\n" +
+                                      "Connection: close\r\n" +
+                                      "\r\n";
+
+            Socket socket = new Socket(server.getHostname(), server.getHttpDefaultPort());
+            socket.setKeepAlive(true);
+
+            BufferedReader bReader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            OutputStream os = socket.getOutputStream();
+
+            os.write(request.getBytes());
+            Thread.sleep(500); // time in-between requests
+            os.write(redirect_request.getBytes());
+
+            // Wait a few seconds to let the connection close if an error occurs
+            Thread.sleep(3000);
+
+            // Read the responses:
+            boolean containsMessage = false;
+            String line;
+            while ((line = bReader.readLine()) != null) {
+                LOG.info(line);
+                if (line.contains("Successfully redirected!")) {
+                    containsMessage = true;
+                }
+            }
+            Assert.assertTrue("The redirect failed!", containsMessage);
+        } finally {
+            if (server != null && server.isStarted()) {
+                server.stopServer();
+            }
+        }
+
+    }
+
+    /**
+     *  ignoreWriteAfterCommit False Scenario.  
+     *  We expect the socket connection to be destroyed
+     *  so we'll see a "Connection reset" exception.
+     *
+     * @throws Exception if something goes horribly wrong
+     */
+    @ExpectedFFDC("com.ibm.wsspi.genericbnf.exception.MessageSentException")
+    @Test
+    public void testIgnoreWriteAfterCommitFalse() throws Exception {
+        try {
+            updateHTTPOptions(false);
+            server.startServer(JSPChannelTest.class.getSimpleName() + ".log");
+
+            String address = server.getHostname() + ":" + server.getHttpDefaultPort();
+            
+            // sendRedirect changed in EE11
+            String page = JakartaEEAction.isEE11OrLaterActive() ? "indexEE11.jsp" : "index.jsp";
+
+            String request = "GET /" + APP_NAME + "/" + page + " HTTP/1.1\r\n" +
+                             "Host: " + address + "\r\n" +
+                             "Keep-Alive: timeout=5, max=200\r\n" +
+                             "\r\n";
+
+            String redirect_request = "GET /" + APP_NAME + "/page2.jsp " + "HTTP/1.1\r\n" +
+                                      "Host: " + address + "\r\n" +
+                                      "Connection: close\r\n" +
+                                      "\r\n";
+
+            Socket socket = new Socket(server.getHostname(), server.getHttpDefaultPort());
```

Comment:
Should we consider resource cleanup for the socket and streams? Maybe a try with resources instead, or a final block that releases regardless of exceptions